ai-quick-actions/model-deployment-tips.md

```diff
@@ -315,19 +315,93 @@ public class RestExample {
 
 ```
 
+### Using `Langchain` with streaming
+
+#### Installation
+The LangChain OCIModelDeployment integration is part of the [`langchain-community`](https://python.langchain.com/docs/integrations/chat/oci_data_science/)  package.  The chat model integration requires **Python 3.9** or newer. Use the following command to install `langchain-community` along with its required dependencies.
+
+```python
+%pip install langgraph "langchain>=0.3" "langchain-community>=0.3" "langchain-openai>=0.2.3" "oracle-ads>2.12"
+```
+
+#### Using Langchain for Completion Endpoint
+```python
+import ads
+from langchain_community.llms import OCIModelDeploymentLLM
+
+# Set authentication through ads
+# Use resource principal are operating within a
+# OCI service that has resource principal based
+# authentication configured
+ads.set_auth("resource_principal")
+
+# Create an instance of OCI Model Deployment Endpoint
+# Replace the endpoint uri and model name with your own
+# Using generic class as entry point, you will be able
+# to pass model parameters through model_kwargs during
+# instantiation.
+llm = OCIModelDeploymentLLM(
+    endpoint="https://modeldeployment.<region>.oci.customer-oci.com/<md_ocid>/predict",
+    model="odsc-llm",
+    streaming=True,
+    model_kwargs={
+        "temperature": 0.2,
+        "max_tokens": 512,
+    },  # other model params...
+)
+
+# Run the LLM
+response = lm.invoke("Who is the first president of United States?")
+
+print(response.content)
+
+```
+
+#### Using Langchain for Chat Completion Endpoint
+```python
+import ads
+from langchain_community.chat_models import ChatOCIModelDeployment
+
+# Use resource principals for authentication
+ads.set_auth(auth="resource_principal")
+
+endpoint = "https://modeldeployment.<region>.oci.customer-oci.com/<md_ocid>/predict"
+# Initialize the chat model with streaming support
+chat = ChatOCIModelDeployment(
+    model="odsc-llm",
+    endpoint=endpoint,
+    # Optionally you can specify additional keyword arguments for the model.
+    max_tokens=1024,
+    # Enable streaming
+    streaming=True
+)
+
+#Invocation
+messages = [
```

Comment:
This example does not work for Mistral model as it does not take system message. Maybe we should mention this as well. There are some discussions: vllm-project/vllm#2112

Reply:
As suggested in the latest conversation in the above thread, I tried the MD endpoint for _Mistral-7B-Instruct-v0.3_, and it's working with the same sample code.

Reply:
Which conversation are your referring to? It does not seem to work for me in with vllm and the `Mistral-7B-Instruct-v0.3` model.